[stable/sentry] Sentry filestore config

damoxc opened the pull request:
<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:

Adds support for configuring Sentry's file storage.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

damoxc committed:
[stable/sentry] provide greater coverage of filestore settings

This stemmed primarily from needing to inject a Google Cloud service account into
the filesystem via a secret. However it seemed sensible if adding support for one
of the additional filestore providers to add support for both.

Signed-off-by: Damien Churchill <damoxc@gmail.com>

damoxc committed:
[sentry/stable] document the change in parameters around filestore

Signed-off-by: Damien Churchill <damoxc@gmail.com>

damoxc committed:
[stable/sentry] Bump chart version to 2.0.0

values.yaml changed in a breaking way, so a major version bump seems prudent.

Signed-off-by: Damien Churchill <damoxc@gmail.com>

damoxc committed:
[stable/sentry] change s3 from commented out to empty block

Signed-off-by: Damien Churchill <damoxc@gmail.com>

damoxc committed:
Merge branch 'master' into sentry-filestore-config

damoxc committed:
Merge branch 'master' into sentry-filestore-config

damoxc committed:
[stable/sentry] correct the values reference for persistentWorkers

Signed-off-by: Damien Churchill <damoxc@gmail.com>

okgolove commented on stable/sentry/values.yaml:
```diff
@@ -109,39 +109,52 @@ service:
+  ## Point this at a pre-configured secret containing a service account. The resulting
+  ## secret will be mounted at /var/secrets/google
+  gcs:
+    credentialsFile: credentials.json
```
Is /var/secrets is a GKE special folder? I've never used it.

damoxc replied:
Whoops, missed that when I updated it, should be `/var/run/secrets` now. Updated.

damoxc committed:
Merge remote-tracking branch 'upstream/master' into sentry-filestore-config

damoxc committed:
[stable/sentry] change /var/secrets to /var/run/secrets

Use the same path format as the Kubernetes default service account to
place the service account key for GCS storage.

Signed-off-by: Damien Churchill <damoxc@gmail.com>

damoxc committed:
[stable/sentry] update path for GCS key in comment in values.yaml

Signed-off-by: Damien Churchill <damoxc@gmail.com>

damoxc committed:
[stable/sentry] correct filesystem path key for mountPath

Signed-off-by: Damien Churchill <damoxc@gmail.com>

damoxc committed:
[stable/sentry] add note in README about the value keys which have changed

Signed-off-by: Damien Churchill <damoxc@gmail.com>
